### PR TITLE
External Sources for Examples

### PR DESCRIPTION
**Pull request by jlmart88**
Implements the spec outlined in #3

### CODE REVIEW

File: test/external-sources-examples.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+```multiple-sources:external_examples/external-template.html
+```
+```multiple-sources:external_examples/external-data.js
+```
+
+or with a wildcard:
+
+```multiple-sources-wildcard:external_examples/*
+```
```
Comment (mpetrovich) [inline test/external-sources-examples.js]: We should add an example with absolute paths too.

File: test/external-sources-examples.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
```
Comment (mpetrovich) [inline test/external-sources-examples.js]: We should add an example with absolute paths too.

**Comment by jlmart88** [comment]
@mpetrovich good idea, I added the test and implementation

File: src/parser.js [hunk 3]
```diff
@@ -196,7 +196,14 @@ function parseDescriptionMarkdown(markdown, component) {
...
 			} else {
 				var externalSourceFilename = `${externalSource}.${language}`;
-				content = fs.readFileSync(path.resolve(componentDir, externalSourceFilename), 'utf8');
+
+				if (externalSourceFilename[0] === '/') {
+					// this is an absolute path, so resolve the source file relative to the runtime directory
+					content = fs.readFileSync(path.resolve(process.cwd(), externalSourceFilename.slice(1)), 'utf8');
+				} else {
+					// otherwise, resolve the source file relative to the component file's directory
+                    content = fs.readFileSync(path.resolve(componentDir, externalSourceFilename), 'utf8');
```
Comment (mpetrovich) [inline src/parser.js]: Mixed spaces/tabs here

File: test/test.js [hunk 4]
```diff
@@ -291,7 +291,19 @@ describe('A processed source code file with external source examples', () => {
...
+            name: 'single-source-absolute',
+            blocks: [
+                {
+                    language: 'html',
+                    hidden: false,
+                    content: `<div id="data">External Example Template</div>\n`
+                },
+            ],
+            options: {},
+        }));
```
Comment (mpetrovich) [inline test/test.js]: ditto

**Comment by mpetrovich** [comment]
This looks great! Don't worry about fixing the indentation issues this time—I can fix them post-merge.

**Pull request merged**